### ISSUE TITLE
Bin entries not being linked to local node_modules/.bin

### ISSUE THREAD

**Issue by nikolakanacki** [yarnpkg/yarn#5713]
I think I've found where the issue is but would like some of the maintainers to give their two cents on it.
If no-one's on it I think I could find some time to submit a pr.

For a more detailed explanation of the issue see: https://github.com/yarnpkg/yarn/issues/4655

This is where we're missing the bin lookup / setup:
https://github.com/yarnpkg/yarn/blob/9e97f50aea60f58042403ff503d757a732756d6a/src/cli/commands/link.js#L47

This is roughly the bin lookup / setup we're missing:
https://github.com/yarnpkg/yarn/blob/9e97f50aea60f58042403ff503d757a732756d6a/src/cli/commands/link.js#L69

**Comment by rally25rs** [comment]
When the command `yarn link whatever` is run, then yeah it looks like when we create the global bin link in `cli/commands/link.js` we should also create a local link. That would happen around line 69 as you linked to above.

I'm a bit worried that when someone next runs a yarn command, like `yarn add left-pad` then it will delete the symlink as extraneous. I haven't tested it, but I suspect it might. If that ends up being the case, then there would need to be a fix somewhere else too, possibly in `package-linker` when it sets up normal bin links for normal dependencies.

**Comment by nikolakanacki** [comment]
@rally25rs  didn't have a time to go through all of that but - does this reads familiar to you / is it related to the possible issue you've pointed out?

https://github.com/yarnpkg/yarn/blob/9e97f50aea60f58042403ff503d757a732756d6a/src/package-linker.js#L291

**Comment by rally25rs** [comment]
Yeah, a little further down in that file: https://github.com/yarnpkg/yarn/blob/9e97f50aea60f58042403ff503d757a732756d6a/src/package-linker.js#L346-L362 it gets all files into a list named `possiblyExtraneous`. It then removes files from that list that match files that come from packages that are currently installed, or are from `yarn link`'d dependencies (as the comment that you linked to indicates). Anything left in that array is deleted as an "extraneous" file.

That code covers files in `node_modules/{package}` directories, but I'm not sure how it handles the links in `node_modules/.bin`. The logic might be different.

**Comment by ctrlplusb** [comment]
Thanks for looking into this all. Are there any known workarounds for now? Manually create the bin links in local node_modules?

**Comment by maelcaldas** [comment]
I was able to workaround this by using [npm-link](https://docs.npmjs.com/cli/link) instead. It creates the link on local node_modules/.bin then you can get back to yarn :-)

**Comment by rleachpear** [comment]
Upvote a fix for this
